### ISSUE TITLE
Don't warn about redundant parentheses around record pattern matching

### ISSUE THREAD

**Issue by anka-213** [ndmitchell/hlint#1282]
The rule that `A {x = y}` binds more tightly than function application `f x` is counter-intuitive and confusing.
```
f A {x = y}
```
looks like `f` is applied to two arguments rather than one.

What I would like is a rule that <s>suggests adding parentheses around </s>`A {x = y}` and disables the warning about redundant parentheses for that case.

**Comment by ndmitchell** [comment]
This is something people disagree on, so I think HLint should remain neutral - in particular the lack of brackets can turn out to be very handy for certain DSLs. That said, given people do disagree, I don't think HLint should suggest _removing_ the bracket. Using the test case:

```
f (A{x=1}) = g (B{x=1})
```

I see that it does suggest removing the brackets on the LHS of the `=`. I think that would be something that should be fixed

**Comment by anka-213** [comment]
To clarify, I intended this whole issue to be opt-in, especially the suggestion to insert the extra parentheses, since people could disagree.
But yes, I would be very happy about not suggesting to remove the brackets on LHS by default and would be very satisfied with just that and no suggestion to insert parentheses and no configuration option about it. That would be better than I hoped for.

**Comment by ndmitchell** [comment]
Actually, trying a recent HLint it doesn't warn on either side with brackets. I'm not a massive fan of opt-in hints because they don't usually get used by very many people at all, so are rarely worth the cost, especially for something like inserting brackets which is fiddly to do right

**Comment by tysonzero** [comment]
@ndmitchell In that case I guess this issue is good to close? Do you happen to know what version of `hlint` changed this? I'm still seeing the redundant brackets suggestion in `3.2.7`.

**Comment by ndmitchell** [comment]
@tysonzero - probably v3.3.1 as part of #1225
